CMakeLists.txt

```diff
@@ -1,52 +1,125 @@
-cmake_minimum_required(VERSION 3.11)
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
 
-project(detray VERSION 0.1 LANGUAGES CXX)
+# Set up the project.
+cmake_minimum_required( VERSION 3.11 )
+project( detray VERSION 0.1 LANGUAGES CXX )
 
-option(CMAKE_WARN_DEPRECATED "Not warning for deprecation in cmake" Off)
-enable_testing()
+# CMake include(s).
+include( CMakeDependentOption )
+include( GNUInstallDirs )
 
-include(CMakeDependentOption)
+# Explicitly set the output directory for the binaries. Such that if this
+# project is included by another project, the main project's configuration would
+# win out.
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY
+   "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}" CACHE PATH
+   "Directory for the built binaries" )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY
+   "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}" CACHE PATH
+   "Directory for the built libraries" )
+set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY
+   "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}" CACHE PATH
+   "Directory for the built static libraries" )
 
 option(DETRAY_ARRAY_PLUGIN "Build std::array math plugin" On)
 option(DETRAY_EIGEN_PLUGIN "Build Eigen math plugin" On)
 option(DETRAY_SMATRIX_PLUGIN "Build ROOT/SMatrix math plugin" Off)
 option(DETRAY_VC_ALGEBRA_PLUGIN "Build vc based simd algebra math plugin" Off)
 option(DETRAY_IO_CSV "Build CSV IO module" On)
-option(DETRAY_IO_JSON "Build JSON IO module" On)
 option(DETRAY_DISPLAY "Build matplot++ based display module" Off)
 option(DETRAY_USE_VECMEM "Build with vecmem" On)
+option( DETRAY_BUILD_TESTING "Build the (unit) tests of Detray"
+   TRUE )
+option( DETRAY_BENCHMARKS "Enable benchmark tests" TRUE )
 
-if(DETRAY_CUSTOM_SCALARTYPE)
```

Comment:
Is the detray custom scalar type gone? There is some testing code that needs this still, I think

Reply:
It is. This **non-cache** variable was a really clumsy thing to have in my mind.

There is a proper cache variable coming with algebra-plugins. We should just modify that on the command line of the detray build if we want to change the scalar type.

To be honest, I actually think that algebra-plugins should not define a scalar type **at all**. I'm not even sure that detray should do that. It should be the client code of algebra-plugins and detray that chooses what scalar type they want to use. Which could, in a crazy world, even be something like CUDA's `__half` type. :stuck_out_tongue:

So I intend to re-organise these projects to only make these scalar type decisions in their test executables. Building `float` and `double` test executables in the same build directory. Without having to do completely separate builds.

Reply:
OK, that's interesting